test/system/liquidation.test.js

```diff
@@ -0,0 +1,127 @@
+const { expect } = require("chai")
+const { BigNumber } = ethers
+const {
+  to1e18,
+  impersonateAccount,
+  resetFork,
+} = require("../helpers/contract-test-helpers")
+const Auction = require("../../artifacts/contracts/Auction.sol/Auction.json")
+
+const describeFn =
+  process.env.NODE_ENV === "system-test" ? describe : describe.skip
+
+// This system test scenario checks the liquidation happy path. It is meant
+// to be executed on Hardhat Network with mainnet forking enabled. At the start,
+// the fork is being reset to the specific starting block which determines the
+// initial test state. This test uses the real tBTC token contract and a
+// deposit (https://allthekeeps.com/deposit/0x55d8b1dd88e60d12c81b5479186c15d07555db9d)
+// which is ready to be liquidated at the starting block. The bidder which
+// takes the offer is also a real account with actual tBTC balance. At the
+// end of the scenario, the risk manager should liquidate the deposit successfully,
+// and 66% of the deposit's bonded amount should land on the signer processor.
+describeFn("System -- liquidation happy path", () => {
+  const startingBlock = 12368838
+  const tbtcTokenAddress = "0x8daebade922df735c38c80c7ebd708af50815faa"
+  const depositAddress = "0x55d8b1dd88e60d12c81b5479186c15d07555db9d"
+  const bidderAddress = "0xa0216ED2202459068a750bDf74063f677613DA34"
+  const auctionLength = 86400 // 24h
+  // deposit lot size is 5 BTC
+  const lotSize = to1e18(5)
+  // signers have bonded 290.81 ETH
+  const bondedAmount = BigNumber.from("290810391624000000000")
+  // 66% of the deposit is exposed on auction in the liquidation moment
+  const bondedAmountPercentage = BigNumber.from("66")
+
+  let tbtcToken
+  let signerBondsSwapStrategy
+  let coveragePool
+  let riskManagerV1
+  let tbtcDeposit
+
+  let bidder
+
+  before(async () => {
+    await resetFork(startingBlock)
+
+    tbtcToken = await ethers.getContractAt("IERC20", tbtcTokenAddress)
+
+    const SignerBondsSwapStrategy = await ethers.getContractFactory(
+      "SignerBondsEscrow"
+    )
+    signerBondsSwapStrategy = await SignerBondsSwapStrategy.deploy()
+    await signerBondsSwapStrategy.deployed()
+
+    const Auction = await ethers.getContractFactory("Auction")
+
+    const masterAuction = await Auction.deploy()
+    await masterAuction.deployed()
+
+    const CoveragePoolStub = await ethers.getContractFactory("CoveragePoolStub")
+    coveragePool = await CoveragePoolStub.deploy()
+    await coveragePool.deployed()
+
+    const RiskManagerV1 = await ethers.getContractFactory("RiskManagerV1")
+    riskManagerV1 = await RiskManagerV1.deploy(
+      tbtcToken.address,
+      coveragePool.address,
+      signerBondsSwapStrategy.address,
+      masterAuction.address,
+      auctionLength
+    )
+    await riskManagerV1.deployed()
+
+    tbtcDeposit = await ethers.getContractAt("IDepositStub", depositAddress)
+
+    bidder = await impersonateAccount(bidderAddress)
+  })
+
```

Comment:
We can also add validation of the initial params after the setup. It will show that certain params have changed because of ex. calling `takeOffer()`. Something like:

```
describe("initial state", () => {
    it("should assert a deposit is in active state", async () => {
      expect(await tbtcDeposit.currentState()).to.equal(5) // Active
    })

    it("should assert an auction does not exist", async () => {
      const auctionAddress = await riskManagerV1.depositToAuction(
        tbtcDeposit.address
      )
      // auctionAddress == ZERO_ADDRESS
    })
  })
  ```

But this is a nice to have I guess..

Reply:
Please see https://github.com/keep-network/coverage-pools/pull/53/commits/2c6ccdba064b98e9a2115957288c9007456156a1